### ISSUE TITLE
h5py/pytables conflict on Windows Anaconda 2.2

### ISSUE THREAD

**Issue by caryan** [ContinuumIO/anaconda-issues#300]
On Windows h5py and pytables have separate and different builds for their hdf5.dll in their site-packages folder.  The second module loaded will then crash when trying to dynamically link against the already loaded dll.  E.g.

```python
Python 2.7.9 |Anaconda 2.2.0 (64-bit)| (default, Dec 18 2014, 16:57:52) [MSC v.1500 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 3.0.0 -- An enhanced Interactive Python.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://binstar.org
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import tables

In [2]: import h5py
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-e81388eeaef8> in <module>()
----> 1 import h5py

C:\Users\qlab\Anaconda\lib\site-packages\h5py\__init__.py in <module>()
     11 _errors.silence_errors()
     12
---> 13 from h5py import _conv
     14 _conv.register_converters()
     15

h5py\h5r.pxd in init h5py._conv (D:\Build\h5py\h5py-2.4.x\h5py\_conv.c:6809)()

h5py\_objects.pxd in init h5py.h5r (D:\Build\h5py\h5py-2.4.x\h5py\h5r.c:2824)()

h5py\_objects.pyx in init h5py._objects (D:\Build\h5py\h5py-2.4.x\h5py\_objects.c:6927)()

ImportError: DLL load failed: The specified procedure could not be found.

In [3]:

```

**Comment by groutr** [comment]
I can confirm on my XP (64bit) machine with a fresh install of Anaconda 2.2.0.  But it only crashes if I load tables and then h5py (the reverse order doesn't produce a crash).

**Comment by caryan** [comment]
It doesn't crash when loading h5py before tables but when pytables actually
tries to do something it will crash.  E.g. try running ``tables.test()``.
Presumably h5py tries to do something with the shared library on import but
pytables defers loading it until some function is called.

**Comment by groutr** [comment]
Something that might be relevant.
http://stackoverflow.com/questions/28333470/use-both-h5py-and-pytables-in-the-same-python-process

**Comment by timtroendle** [comment]
Can reproduce that on our Win64 machines. 

The problem seems to lie in build ``0`` of ``h5py 2.4.0``. This does not happen with version ``2.3.x``. So a possible workaround is to uninstall ``2.4.0`` and install ``2.3.x``.

**Comment by timtroendle** [comment]
Build ``0`` of version ``2.5.0`` of ``h5py`` shows the same problems for me.

**Comment by caryan** [comment]
This appears to have been fixed in Anaconda 2.3
